src/Data/Text/Array.hs

```diff
@@ -78,6 +78,17 @@ import GHC.ST (ST(..), runST)
 import GHC.Word (Word16(..))
 import Prelude hiding (length, read)
 
+#if MIN_VERSION_base(4,16,0)
+import GHC.Base ( word16ToWord#, wordToWord16# )
+#else
+import GHC.Prim (Word#)
+word16ToWord#, wordToWord16# :: Word# -> Word#
```

Comment:
Let's put these into `Data.Text.Internal.PrimCompat`, next to `Data.Text.Internal.ByteStringCompat`.